sdl-core/src/main/scala/io/smartdatalake/workflow/action/customlogic/CustomFileCreator.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Smart Data Lake - Build your data lake the smart way.
+ *
+ * Copyright © 2019-2020 Schweizerische Bundesbahnen SBB (<https://www.sbb.ch>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.smartdatalake.workflow.action.customlogic
+
+import java.io.ByteArrayInputStream
+
+import io.smartdatalake.util.hdfs.HdfsUtil
+import io.smartdatalake.util.misc.CustomCodeUtil
+import org.apache.spark.sql.SparkSession
+
+trait CustomFileCreator extends Serializable {
+
+  /**
+   * This function creates a [[ByteArrayInputStream]] based on custom code.
+   *
+   * @param session the Spark Session
+   * @param config  input config of the action
+   * @return a stream containing the custom file data as a sequence of bytes
+   */
+  def exec(session: SparkSession, config: Map[String, String]): ByteArrayInputStream
```

Comment:
I suggest to change the interface to return an InputStream and not a ByteArrayInputStream. This is more general and doesn't force the implementation to materialize the data as array.
```suggestion
  def exec(session: SparkSession, config: Map[String, String]): InputStream
```

Reply:
All good points. I changed the signature to use InputStream. Regarding being able to write to a CustomFileDataObject: Would be nice, but there are some questions that need to be addressed before doing so.